spikesorters/launcher.py

```diff
@@ -137,9 +147,15 @@ def run_sorters(sorter_list, recording_dict_or_list, working_folder, sorter_para
                 else:
                     raise (ValueError('mode not in raise, overwrite, keep'))
             params = sorter_params.get(sorter_name, {})
-            task_list.append((recording, sorter_name, output_folder, grouping_property, verbose, params, raise_error))
-
-    if engine is None or engine == 'loop':
+            if need_serialize:
+                print(engine)
+                assert recording.is_dumpable, 'run_sorters(engine=... ) if engine is not "loop" then recording have to be dumpable'
```

Comment:
instead use `recording.check_if_dumpable()`, because it looks at the entire nested structure. `recording.is_dumpable` is only the first layer